plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/QueryInsightService.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.Nullable;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.lifecycle.AbstractLifecycleComponent;
+import org.opensearch.plugin.insights.core.exporter.QueryInsightExporter;
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Service responsible for gathering, analyzing, storing and exporting data related to
+ * search queries, based on certain dimensions.
+ *
+ * @param <R> The type of record that stores in the service
+ * @param <S> The type of Collection that holds the aggregated data
+ * @param <E> The type of exporter that exports the aggregated and processed data
+ *
+ * @opensearch.internal
+ */
+public abstract class QueryInsightService<R extends SearchQueryRecord<?>, S extends Collection<R>, E extends QueryInsightExporter<R>>
```

Comment:
@Same. Let's try to follow consistent naming. Either change it to `QueryInsight` in all places or `QueryInsights`.

Reply:
Good catch! let me rename it.